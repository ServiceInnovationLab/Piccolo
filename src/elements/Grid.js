import styled from 'styled-components';

const Grid = styled.div`
  margin: 30px 0;
  @media(min-width: 1024px) {
    display: flex;
  }
`;

export const Column = styled.div`
  padding: 0 40px 10px 40px;
  border-top: 1px solid #000;
  padding-top: 40px;
  margin-top: 50px;
  h2:first-child {
    margin-top: 0;
  }
  h2 + p {
    margin-top: 20px;
  }
  h2 ~ p {
    margin-bottom: 0;
  }
  p:last-child {
    margin-bottom: 0;
    padding-bottom: 0;
  }
  @media(min-width: 1024px) {
    flex: 1;
    border-left: 1px solid #000;
    border-top: none;
    padding-top: 0;
  }
`;

export default Grid;

