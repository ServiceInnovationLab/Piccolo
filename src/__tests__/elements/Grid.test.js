import React from 'react';
import Grid, { Column } from '../../elements/Grid';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

test('Grid renders correctly', () => {
  const tree = renderer.create(<Grid />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('margin', '30px 0');
});

test('Column renders correctly', () => {
  const tree = renderer.create(<Column />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('padding', '0 40px 10px 40px');
  expect(tree).toHaveStyleRule('border-top', '1px solid #000');
  expect(tree).toHaveStyleRule('padding-top', '40px');
  expect(tree).toHaveStyleRule('margin-top', '50px');
});
