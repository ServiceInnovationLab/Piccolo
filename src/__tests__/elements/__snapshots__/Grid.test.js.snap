// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column renders correctly 1`] = `
.c0 {
  padding: 0 40px 10px 40px;
  border-top: 1px solid #000;
  padding-top: 40px;
  margin-top: 50px;
}

.c0 h2:first-child {
  margin-top: 0;
}

.c0 h2 + p {
  margin-top: 20px;
}

.c0 h2 ~ p {
  margin-bottom: 0;
}

.c0 p:last-child {
  margin-bottom: 0;
  padding-bottom: 0;
}

@media (min-width:1024px) {
  .c0 {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
    border-left: 1px solid #000;
    border-top: none;
    padding-top: 0;
  }
}

<div
  className="c0"
/>
`;

exports[`Grid renders correctly 1`] = `
.c0 {
  margin: 30px 0;
}

@media (min-width:1024px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div
  className="c0"
/>
`;
